### PR TITLE
Create code of conduct

### PR DESCRIPTION
**Pull request by Kalashnikkov**
GitHub Issue (If applicable): #7 

## PR Type

What kind of change does this PR introduce?

- Documentation content changes

## What is the current behavior?

No code of conduct.

## What is the new behavior?

New code of conduct based upon an existing template used by many.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current upstream repo
- [ ] Added new tests to the test suite for the feature
- [ ] Tested the full test suite, and ensured that the feature does not break current build.
- [x] Docs have been added/updated which fit (for bug fixes / features)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)

## Other information

This may need to be updated as we see fit.

### CODE REVIEW

File: .github/code_of_conduct.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+* Trolling, insulting/derogatory comments, and personal or political attacks
+* Public or private harassment
+* Publishing others' private information, such as a physical or electronic
+ address, without explicit permission
+* Other conduct which could reasonably be considered inappropriate in a
+ professional setting
+
+## As a SE701 Student in Group 5
+
+We all expect to have great communication skills, which includes the use of either Discord or Messenger
```
Comment (Matteas-Eden) [inline .github/code_of_conduct.md]: Perhaps change to `We are all expected to...`

File: .github/code_of_conduct.md [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+(Whichever suits best for you). This also includes in the case of being unable to participate for
```
Comment (Matteas-Eden) [inline .github/code_of_conduct.md]: Perhaps change to `This includes cases of [...] for periods of time, in which case letting the team know aids in any future planning that may be required.`

File: .github/code_of_conduct.md [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+* Other conduct which could reasonably be considered inappropriate in a
+ professional setting
+
+## As a SE701 Student in Group 5
+
+We all expect to have great communication skills, which includes the use of either Discord or Messenger
+(Whichever suits best for you). This also includes in the case of being unable to participate for 
+a time being, letting the team know will aid in any future planning that may be done.
+
+We all expect to be actively participating in construction of this project. It is noted that we are all
```
Comment (Matteas-Eden) [inline .github/code_of_conduct.md]: Again, maybe change to `We are all expected to...`

File: .github/code_of_conduct.md [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+students and we have other commitments. However, any contribution is good contribution, as long as you
```
Comment (Matteas-Eden) [inline .github/code_of_conduct.md]: Maybe modify to read `good contribution, so long as...`

File: .github/code_of_conduct.md [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+further defined and clarified by project maintainers.
+
+## Conflict Resolution
+
+As the project grows there *will* be some conflict and debate as development continues.
+However, this is not an excuse for *ANY* type of bullying. We encourage healthy discussion
+as this helps everyone in furthering their own skill and the project's development.
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be
+reported by contacting Kelly at k.blincoe@auckland.ac.nz. All
```
Comment (Matteas-Eden) [inline .github/code_of_conduct.md]: Maybe surround Kelly's email with `

File: .github/code_of_conduct.md (hunk 3 above)
Comment (PreetPatel) [reply .github/code_of_conduct.md]: Agree.. this sounds better

File: .github/code_of_conduct.md (hunk 5 above)
Comment (PreetPatel) [reply .github/code_of_conduct.md]: Alternatively use `[mailto](mailto:k.blincoe@auckland.ac.nz)`

Comment (Kalashnikkov) [reply .github/code_of_conduct.md]: It automatically already sets it up as a hyperlink to `mailto:`. What are our thoughts on changing this to a blocked section or leaving it as is?

Comment (Kalashnikkov) [reply .github/code_of_conduct.md]: image


cc: @PreetPatel  @Matteas-Eden

Comment (Matteas-Eden) [reply .github/code_of_conduct.md]: My suggestions are non-blocking

Comment (Kalashnikkov) [reply .github/code_of_conduct.md]: blocked as in `block`

Comment (jordansimsmith) [reply .github/code_of_conduct.md]: > It automatically already sets it up as a hyperlink to `mailto:`. What are our thoughts on changing this to a blocked section or leaving it as is?

Fine as is, blocks are not linkable afaik

**Pull request merged**